content/blogposts/2023/guest-post-arafat-khan-php-python.md

```diff
@@ -0,0 +1,254 @@
+  caption="PHP Variable fixes with Cody"
+/>
+
+The other thing that I really enjoyed was the improved speed of coding.
+
+Typically, the coding process goes like this:
+
+1. Conceptualize
+2. Draft an imperfect code
+3. Refine
+4. Comment
+    
+However, with Cody, I experienced a much faster approach:
+
+1. Conceptualize
+2. Comment
+3. Generate a near-perfect code
+4. Refine
+
+<Figure
+  src="https://storage.googleapis.com/sourcegraph-assets/blog/guest-post-arafat-khan-php-python/image_6.jpg"
+  caption="Couldn’t resist adding an AI image to show the contrast of two Coders"
+/>
+
+Sometimes, I would just describe what the function is “supposed” to do in a comment, and just like magic, Cody would write it for me.
+
+The above example is a classic Cody’s autocomplete feature. It works surprisingly well for all kinds of code completion tasks, especially when your repo already has a reasonable amount of code to give it context. This autocomplete system is grounded in a complex foundation, and for those interested in its intricacies, you can delve deeper [here](https://about.sourcegraph.com/blog/the-lifecycle-of-a-code-ai-completion).
+
+All these features, such as_ autocomplete, explanation, code fixes, and testing,_ made my job as a programmer a lot easier. I was able to build the building blocks of my code, and then, inch by inch, I was getting closer to completing the entire project. As I was building more and more of this codebase, a question that crossed my mind was:
+
+## Why use Cody when you have GPT-4?
+
+I like the speed and convenience of Cody but I can’t quite tell why using the Vscode extension is so much better than using GPT 4 if I just wanted to ask code questions. This is where the real magic of Cody comes in. Cody can answer questions about your entire codebase and sift through all the different parts to find the most relevant information pertaining to your question. It uses the [Claude 100K model](https://www.anthropic.com/index/100k-context-windows) (soon to be the [200K model](https://www.anthropic.com/index/claude-2-1)) so it has the capacity to fit in so much more context than my copy-pasting hacks ever could. Take, for example, this question
+
+<Figure
+  src="https://storage.googleapis.com/sourcegraph-assets/blog/guest-post-arafat-khan-php-python/image_7.png"
+/>
+
+I asked a question here that touches 11 different files, each of which would easily exceed the context window of GPT 4 (Never mind the copy-pasting hassle). It sifts through all the files and shows me how they are interconnected and what calls touch which parts. This to me, is the most important feature of Cody: the ability to answer questions that are deeply tangled in the complexity of the codebase. It helps me answer questions that I simply can’t find through a simple copy-paste because the real problem is sometimes_ I don’t even know where to look_ when I am dealing with very large codebases.
```

Comment:
```suggestion
I asked a question here that touches 11 different files, each of which would easily exceed the context window of GPT 4 (Never mind the copy-pasting hassle). It sifts through all the files and shows me how they are interconnected and what calls touch which parts. This to me, is the most important feature of Cody: the ability to answer questions that are deeply tangled in the complexity of the codebase. It helps me answer questions that I simply can’t find through a simple copy-paste because the real problem is sometimes _I don’t even know where to look_ when I am dealing with very large codebases.
```

Reply:
<img width="289" alt="image" src="https://github.com/sourcegraph/about/assets/11155207/4f71660d-b44e-4b6d-bfbb-15512276c6e9">